### PR TITLE
[Converter] add option about supported media type - @open sesame 04/05 12:32

### PR DESCRIPTION
**Pull request by jaeyun-jung**
1. add meson option to disable video support.
tensor-converter will not get the video if video-support disabled.
2. add macros for media format and info to disable audio/video stream.

```
if both audio and video disabled,

Pad Templates:
  SINK template: 'sink'
    Availability: Always
    Capabilities:
      text/x-raw
                 format: utf8
      application/octet-stream

  SRC template: 'src'
    Availability: Always
    Capabilities:
      other/tensor
              framerate: [ 0/1, 2147483647/1 ]
```

Signed-off-by: Jaeyun Jung <jy1210.jung@samsung.com>

### CODE REVIEW

File: gst/nnstreamer/tensor_converter/tensor_converter.c [hunk 1]
```diff
@@ -250,8 +254,10 @@ gst_tensor_converter_class_init (GstTensorConverterClass * klass)
 
   /* set sink pad template */
   pad_caps = gst_caps_new_empty ();
+#ifndef NO_VIDEO
   /* video */
   gst_caps_append (pad_caps, gst_caps_from_string (VIDEO_CAPS_STR));
```
Comment (myungjoo) [inline gst/nnstreamer/tensor_converter/tensor_converter.c]: In later versions, as suggested in today's chat, let's do
```
#ifdef NO_VIDEO
#define gst_caps_append (a, b) do { } while(0)
#else
#include <blahblah.h>
#endif
```
at the first lines of the code (or at converter header) to "centralize" #ifdef/#endif and don't use #ifdef/#endif in the middle of the code.

File: gst/nnstreamer/tensor_converter/tensor_converter.c [hunk 2]
```diff
@@ -44,7 +44,9 @@
 
 #include <string.h>
 #include "tensor_converter.h"
+#ifndef NO_VIDEO
```
Comment (leemgs) [inline gst/nnstreamer/tensor_converter/tensor_converter.c]: Looks good. BTW, where can I find a unit test and test case in case of NO_VIDEO?

**Comment by jaeyun-jung** [comment]
Errors
[gst/nnstreamer/tensor_converter/tensor_converter.c:1401]: (error) Uninitialized variable: info
[gst/nnstreamer/tensor_converter/tensor_converter.c:1367]: (error) Uninitialized variable: info

both initialed with gst_audio_info_init (&info) and gst_video_info_init (&info)

File: gst/nnstreamer/tensor_converter/tensor_converter.c (hunk 1 above)
Comment (jaeyun-jung) [reply gst/nnstreamer/tensor_converter/tensor_converter.c]: add macros to set pad template and check media types

File: gst/nnstreamer/tensor_converter/tensor_converter.c (hunk 2 above)
Comment (jaeyun-jung) [reply gst/nnstreamer/tensor_converter/tensor_converter.c]: Now the features NO_VIDEO/NO_AUDIO are removed in tensor_converter.c
Defined macros in media type header and uses these.

When enabled no-video definition, most of the testcases would be broken.

**Comment by kparichay** [comment]
> Errors
> [gst/nnstreamer/tensor_converter/tensor_converter.c:1401]: (error) Uninitialized variable: info
> [gst/nnstreamer/tensor_converter/tensor_converter.c:1367]: (error) Uninitialized variable: info
> 
> both initialed with gst_audio_info_init (&info) and gst_video_info_init (&info)

Newer version of cppcheck (1.87) has been installed. Please re-build.

**Comment by jaeyun-jung** [comment]
To avoid cpp checker error, added memset before init video and audio info.

**Comment by myungjoo** [comment]
PTAL: @jijoongmoon @leemgs

**Pull request merged**